### PR TITLE
Cancel appointment logic

### PR DESCRIPTION
**Pull request by SimWPEric**
1. Create the CancelCommand, which serves as the equivalent of
`delete` for Appointments.
2. Set the COMMAND_WORD to `cancel` to prevent conflicts with the
existing `DeleteCommand`.

### CODE REVIEW

**Review by hcs1203 — APPROVED** [review]

File: src/main/java/seedu/address/logic/commands/CancelCommand.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+    public static final String MESSAGE_CANCEL_APPOINTMENT_SUCCESS = "Cancelled Appointment: %1$s";
+
+    private final Index targetIndex;
+
+    public CancelCommand(Index targetIndex) {
+        this.targetIndex = targetIndex;
+    }
+
+    @Override
```
Comment (quzhetao01) [inline src/main/java/seedu/address/logic/commands/CancelCommand.java:34]: execute methods LGTM, just need to test it out once an appointment can be added

File: src/main/java/seedu/address/logic/commands/CancelCommand.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        }
+
+        Appointment appointmentToDelete = lastShownList.get(targetIndex.getZeroBased());
+        // TODO model.deleteAppointment(appointmentToDelete);
+        return new CommandResult(String.format(MESSAGE_CANCEL_APPOINTMENT_SUCCESS,
+                Messages.format(appointmentToDelete)));
+    }
+
+    @Override
+    public boolean equals(Object other) {
```
Comment (quzhetao01) [inline src/main/java/seedu/address/logic/commands/CancelCommand.java:50]: Test cases for this in the future as well

**Review by darrentfy — APPROVED** [review]
LGTM, nice that test cases are extensive

**Review by quzhetao01 — APPROVED** [review]
LGTM! There is a need to add some test cases in the future

**Pull request merged**